File: Backend/src/main/java/com/DevPulse/service/AuthService.java
package com.DevPulse.service;

import com.DevPulse.dto.LoginRequest;
import com.DevPulse.dto.LoginResponse;
import com.DevPulse.dto.SignupRequest;
import com.DevPulse.dto.UserResponse;

public interface AuthService {

    UserResponse signup(SignupRequest request);

    LoginResponse login(LoginRequest request);
}

File: Frontend/devpulse-frontend/src/services/authService.ts
import api from "../api/axiosConfig";

import { LoginRequest, LoginResponse } from "../interfaces/Login";
import { SignupRequest } from "../interfaces/Signup";

export const signup = async (data: SignupRequest): Promise<void> => {
    
    const response = await api.post("/auth/signup",data);

    return response.data;
}

export const login = async(data:LoginRequest): Promise<LoginResponse> => {
    const response = await api.post("/auth/login",data);
    console.log(response.data);
    
    return response.data;
}